applications/vanilla/controllers/api/DiscussionsApiController.php

```diff
@@ -645,4 +667,68 @@ public function put_bookmark($id, array $body) {
         $result = $this->discussionByID($id);
         return $out->validate($result);
     }
+
+    /**
+     * Set canonical url for a discussion.
+     *
+     * @param int $id The ID of the discussion.
+     * @param array $body The request body.
+     *       Ex: ["canonicalUrl" => "https://mydomain.com/some+path/"]
+     * @throws NotFoundException If unable to find the discussion.
+     * @return array
+     */
+    public function put_canonicalUrl($id, array $body) {
+        $this->permission('Garden.SignIn.Allow');
```

Comment:
We should really get some stricter permissions on this endpoint.

Reply:
That is kind the same way we have for discussions `post()`
https://github.com/vanilla/vanilla/blob/29ab4e729ac0b14fb19998a4f3ce923cd5591d7d/applications/vanilla/controllers/api/DiscussionsApiController.php#L596

Reply:
You're absolutely right. I skimmed over the permission checks you have, further down. Sorry!